lib/WebpackOptionsDefaulter.js

```diff
@@ -166,25 +170,16 @@ class WebpackOptionsDefaulter extends OptionsDefaulter {
 		this.set("node.__filename", "mock");
 		this.set("node.__dirname", "mock");
 
-		this.set(
-			"performance",
-			"make",
-			options => (isProductionLikeMode(options) ? false : undefined)
-		);
-		this.set("performance", "call", value => {
-			if (typeof value === "boolean") {
-				return value;
+		this.set("performance", "make", options => {
+			if (isWebLikeTarget(options) && isProductionLikeMode(options)) {
+				return {};
 			} else {
-				return Object.assign({}, value);
```

Comment:
The `"call"` with `Object.assign({}, value)` is important to clone the options. You have to keep it.

But merging these rules is great. Maybe by keeping the `"call"` rule and adding

``` js
if(value === undefined && (!isProductionLikeMode(options) || !isWebLikeTarget(options)) return false;
```